compiler/plugins/target/AMD-AIE/aie/AMDAIECoreToStandard.cpp

```diff
@@ -251,6 +257,28 @@ struct AMDAIECoreToStandardPass : mlir::OperationPass<ModuleOp> {
     if (failed(applyPartialConversion(m, target, std::move(patterns))))
       return signalPassFailure();
 
+    // Assert that cores are isolated
+    {
```

Comment:
You could put it in a utility function inside the CoreOp itself so it can be reused in different places. But I wouldn't spend effort on that for AIE ops and wait for it until we solely rely on AMDAIE ops.